### DISCUSSION TITLE
Is There a Way to Programmatically Access Audit Log Records?

### DISCUSSION THREAD

**Discussion by asluis3** [apache/airflow#34142] [Q&A] [answered]
Hi everyone,

I am using Airflow v2.5.3 and I'm trying to access Audit Log contents without having to query the database. I'm reading from Audit Logs from a Listener and having to query the database for every event does not seem efficient (and it also throws HA lock errors). Thank you for the help!

**Comment by Taragolis** [comment, marked as answer]
Since audit logs stored in DB you need to query DB, however DB structure could changed from version to version so I would suggest to use [REST API EventLog endpoint](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/get_event_logs)
